test/unit/reductions/grid_search/test_grid_generator.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from fairlearn.reductions._grid_search import _GridGenerator
+from fairlearn.reductions import DemographicParity
+
+from test.unit.reductions.grid_search import _utilities
+
+
+@pytest.mark.parametrize("grid_size", [5, 6, 10, 11, 100, 101, 102])
+@pytest.mark.parametrize("grid_limit", [1.0, 2.0, 4.0, 10.0])
+def test_grid_generator(grid_size, grid_limit):
```

Comment:
'_1d' since I believe that the grid generator can do multi-d as written.

Reply:
Correct! I should probably add a similar test case for that, and maybe other constraints